### DISCUSSION TITLE
Remote workflows seems to send empty strings for non-required arguments

### DISCUSSION THREAD

**Discussion by caoilte** [community/community#63920] [Actions]
### Select Topic Area

Bug

### Body

I had a confusing day yesterday debugging a GHE Workflow that failed when run remotely but not when run locally. The reason was obvious when I finally found it but is a really nasty sharp edge in the system.

Basically, from what i can tell, when you run a workflow locally it will fallback to the default argument of a non required input. However, running a workflow remotely seems to send empty strings for every argument that isn't provided as it was not falling back to the default.

**Comment by oftharsh** [comment]
If you like it please mark it as answer.
It looks like you're describing a bug you encountered while working with GitHub Actions workflows. The issue seems to be related to how inputs with default values are handled differently when a workflow is run locally versus when it's run remotely on GitHub's servers.

In your description, you mentioned that:

1. When running a workflow locally, it correctly falls back to the default value of a non-required input if the input is not provided.

2. However, when running the same workflow remotely on GitHub's servers, the workflow is sending empty strings for every argument that isn't provided, instead of falling back to the default value.

This inconsistency between local and remote behavior can indeed be confusing and lead to unexpected issues during debugging and development. It's important for workflows to behave consistently regardless of where they are executed.

To address this bug, you might want to consider the following steps:

1. Reproduce the Issue: Try to reproduce the issue in a controlled environment, if possible. Create a minimal example workflow that showcases the inconsistent behavior. This will help you isolate the problem and provide a clear demonstration to GitHub's support team.

2. GitHub Support: Reach out to GitHub's support or community forums to report the issue. Provide the detailed information you've shared here, including the description of the bug, the steps to reproduce it, and any relevant code snippets or configuration files.

3. Check for Updates: Check if there have been any recent updates or changes to GitHub Actions that might have addressed this issue. It's possible that the bug has already been identified and fixed in a newer version.

4. Provide Feedback: If you can, provide feedback on the GitHub Actions documentation or any related GitHub repositories. This can help raise awareness about the issue and potentially lead to a quicker resolution.

5. Workaround: In the meantime, you might need to find a workaround to ensure consistent behavior between local and remote workflow execution. This could involve handling inputs and defaults in a way that accommodates both scenarios.

Remember that GitHub's support and community are valuable resources for addressing such issues. They can provide insights, guidance, and potentially help with resolving the bug or providing a workaround until it's fixed.

**Comment by Jaber-Saed** [comment]
The error message "machine" not found typically indicates that the module named "machine" is not available in your current Python environment. The "machine" module is commonly used in MicroPython, a variant of Python optimized for microcontrollers like the Raspberry Pi Pico.

To resolve this issue, follow these steps:

    Check MicroPython Installation:
    Make sure you have MicroPython installed on your Raspberry Pi Pico. You need to install MicroPython firmware onto the Pico before you can use the "machine" module. You can find the MicroPython firmware for Pico on the official MicroPython website.

    Install Required Modules:
    If you have MicroPython installed but still encounter the error, ensure that you're importing the "machine" module correctly in your code. The line to import the "machine" module should look like this:

    python

    import machine

    Verify Environment:
    Double-check that you are running your code in a MicroPython environment. If you're using a regular Python environment, the "machine" module won't be available. You might need to use a specific MicroPython editor or terminal for your Raspberry Pi Pico.

    Check Syntax and Spelling:
    Verify that you've spelled "machine" correctly and that there are no typos in your code. Python is case-sensitive, so ensure that the casing is accurate.

    Use the Correct Shell:
    When working with MicroPython on the Raspberry Pi Pico, you often use a special shell or editor that's compatible with MicroPython. Make sure you're using the correct shell or interface to execute your code.

    Reflash Firmware:
    If you're still encountering issues, consider re-flashing the MicroPython firmware onto your Raspberry Pi Pico. This can sometimes resolve any issues related to missing modules.

    Consult Documentation and Resources:
    Refer to official MicroPython documentation, forums, and tutorials specific to your Raspberry Pi Pico setup. They can provide guidance on setup, troubleshooting, and using MicroPython modules.

    Check YouTube Video Details:
    Review the YouTube video details to ensure you're following the instructions accurately. Sometimes, videos might miss a step or there could be specific setup details mentioned in the video description or comments.

If you continue to face issues after trying these steps, provide more specific details about the steps you've taken, your code, and any error messages you're encountering. This will help the community provide more targeted assistance.

**Reply by caoilte** [reply]
This has been posted on the wrong discussion thread i think.
